src/components/BarChart/Chart.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react'
+import Svg, { Text, Rect } from 'react-native-svg'
+import { PanResponder, Animated } from 'react-native'
+import { maxBy, clamp, max } from 'lodash'
+import Haptic from 'react-native-haptic-feedback'
+import { ChartData } from './types'
+
+// TODO
+// scale gap between bars
+// if all downs are 0, no gap
+// split the difference between the gaps when finding the data target
+// min height, circle
+// animate in
+
+type Props = {
+  width: number
+  height: number
+  data: ChartData[]
+  onFocus: (data: ChartData | null) => void
+}
+
+const BarChart = ({ width, height, data, onFocus }: Props) => {
+  const [focusedBar, setFocusedBar] = useState(null)
+
+  // trigger haptic feedback when the focused bar changes
+  useEffect(() => {
+    if (focusedBar) {
+      Haptic.trigger('impactMedium')
+    }
+    onFocus(focusedBar)
+  }, [focusedBar, onFocus])
+
+  // SVG maths
+  const barWidth = width / (data.length + data.length - 1)
+  const maxUp = maxBy(data, 'up')?.up || 0
+  const maxDown = maxBy(data, 'down')?.down || 0
+  const maxBarHeight = maxUp + barWidth / 1.5 + maxDown
+  const vScale = (height - 20) / maxBarHeight
+  const minBarHeight = barWidth
```

Comment:
Is minBarHeight supposed to be barWidth?

Reply:
yeah it makes them round below that threshold

Reply:
Ok just wanted to be sure